.github/workflows/docker-build-push.yml

```diff
@@ -0,0 +1,98 @@
+  GCE_INSTANCE: deploy-pudl-vm
+  GCE_INSTANCE_ZONE: us-central1-a
+
+jobs:
+  push_to_registry:
+    name: Build Docker image and push to Docker Hub
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Docker Metadata
+        id: docker_metadata
+        uses: docker/metadata-action@v4.0.1
+        with:
+          images: catalystcoop/pudl-etl
+          flavor: |
+            latest=auto
+          # edge is for testing purposes
+          tags: |
+            type=ref,event=branch
+            type=ref,event=tag
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2.0.0
+
+      - name: Login to DockerHub
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@v2.0.0
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build image and push to Docker Hub
+        uses: docker/build-push-action@v3.0.0
+        with:
+          context: .
+          file: docker/Dockerfile
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.docker_metadata.outputs.tags }}
+          labels: ${{ steps.docker_metadata.outputs.labels }}
+
+  deploy_to_gce:
+    name: Deploy the image to a GCE VM
+    runs-on: ubuntu-latest
+    needs: push_to_registry
+
+    # Add "id-token" with the intended permissions.
+    permissions:
+      contents: "read"
+      id-token: "write"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      # Authentication via credentials json
+      - id: "auth"
+        uses: "google-github-actions/auth@v0"
+        with:
+          credentials_json: "${{ secrets.GCE_SA_KEY }}"
+
+      # Setup gcloud CLI
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v0
+
+      # Start the VM
+      - name: Start the deploy-pudl-vm
+        run: gcloud compute instances start "$GCE_INSTANCE" --zone="$GCE_INSTANCE_ZONE"
+
+      # Deploy PUDL image to GCE
+      - name: Deploy
+        run: |-
+          gcloud compute instances update-container "$GCE_INSTANCE" \
+            --zone "$GCE_INSTANCE_ZONE" \
+            --container-image "docker.io/catalystcoop/pudl-etl:${{github.ref_name}}" \
+            --container-command "conda" \
+            --container-arg="run" \
+            --container-arg="--no-capture-output" \
+            --container-arg="-p" \
+            --container-arg="/home/catalyst/env" \
+            --container-arg="bash" \
+            --container-arg="./docker/pudl_etl.sh" \
+            --container-env-file="./docker/.env" \
+            --container-env GITHUB_SHA=${GITHUB_SHA} \
+            --container-env GITHUB_REF=${{github.ref_name}} \
+            --container-env API_KEY_EIA=${{ secrets.API_KEY_EIA }} \
+            --container-env SLACK_TOKEN=${{ secrets.PUDL_DEPLOY_SLACK_TOKEN }}
```

Comment:
Yeah, it's kind of a nightmare. `gcloud` [requires](https://cloud.google.com/compute/docs/containers/configuring-options-to-run-containers#passing_arguments_to_container_entrypoint_command) a separate line for each command arg. We could also make the `pudl_etl.sh` command the dockerfile entry point instead of the unit tests.

Reply:
Running the unit tests was really just a placeholder to make sure stuff was running. It seems like this is the main use case for the container and it totally seems legit to swap in the new script as the entrypoint.